server/odc-service/src/main/java/com/oceanbase/odc/service/integration/oauth2/Oauth2StateManager.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.integration.oauth2;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.oauth2.core.endpoint.OAuth2ParameterNames;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.oceanbase.odc.core.shared.Verify;
+import com.oceanbase.odc.core.shared.constant.OdcConstants;
+import com.oceanbase.odc.service.common.response.SuccessResponse;
+import com.oceanbase.odc.service.common.util.WebRequestUtils;
+import com.oceanbase.odc.service.dispatch.RequestDispatcher;
+import com.oceanbase.odc.service.info.OdcInfoService;
+
+import lombok.SneakyThrows;
+
+@Component
+public class Oauth2StateManager {
+
+    private final Cache<String, Map<String, String>> STATE_PARAM_CACHE = CacheBuilder.newBuilder().maximumSize(1000)
+            .expireAfterWrite(10, TimeUnit.MINUTES)
+            .build();
+
+    @Autowired
+    private RequestDispatcher requestDispatcher;
+
+    @SneakyThrows
+    public void addState(String state, String key, String value) {
+        Map<String, String> stringStringMap = STATE_PARAM_CACHE.get(state, HashMap::new);
```

Comment:
addState or setStateParameter?

if state mapping to multiple parameters, use ConcurrentHashMap instead. or line 54 may result into concurrent conflict?

Reply:
change to setStateParameter.
For the same state, it‘s processed in the same request, which ensures that no other threads compete, so there will be no concurrency issues.